contracts/BridgeExecutorBase.sol

```diff
@@ -121,6 +121,14 @@ abstract contract BridgeExecutorBase is IBridgeExecutor {
 
   function receiveFunds() external payable {}
 
+
+  /// @inheritdoc IBridgeExecutor
+  function updateGuardian(address guardian) external override onlyThis {
+    emit GuardianUpdate(_guardian, guardian);
```

Comment:
Any reason to emit before state modification @stevenvaleri ?

Reply:
It's to include the old and the new without adding another variable

Reply:
Makes sense. I guess it is okay, as they are clearly atomic in all senses, being just an internal operation